netbox/core/views.py

```diff
@@ -642,11 +648,145 @@ def get(self, request):
+            payload['page'] = page
+            next_page = session.get(url, params=payload).json()
+            yield next_page
+
+    for page in get_pages():
+        for data in page['data']:
+
+            versions = []
+            versions.append(data['release_latest'])
+            versions.extend(data['release_recent_history'])
+            if data['slug'] in plugins:
+                plugins[data['slug']]['is_local'] = False
+                plugins[data['slug']]['is_certified'] = data['release_latest']['is_certified']
+                plugins[data['slug']]['description_short'] = data['description_short']
+            else:
+                plugins[data['slug']] = {
+                    'slug': data['slug'],
+                    'config_name': data['config_name'],
+                    'name': data['title_short'],
+                    'title_long': data['title_long'],
+                    'tag_line': data['tag_line'],
+                    'description_short': data['description_short'],
+                    'author': data['author']['name'] or _('Unknown Author'),
+                    'created': datetime_from_timestamp(data['created_at']),
+                    'updated': datetime_from_timestamp(data['updated_at']),
+                    'is_local': False,
+                    'is_installed': False,
+                    'is_certified': data['release_latest']['is_certified'],
+                    'is_community': not data['release_latest']['is_certified'],
+                    'versions': versions,
+                }
+
+    return plugins
+
+
+def get_plugins():
+    if plugins := cache.get('plugins-catalog-feed'):
+        return plugins
+
+    plugins = {}
+    plugins = get_local_plugins(plugins)
+    plugins = get_catalog_plugins(plugins)
+
+    cache.set('plugins-catalog-feed', plugins, 3600)
+    return plugins
+
+
+class PluginListView(UserPassesTestMixin, View):
+
+    def test_func(self):
+        return self.request.user.is_staff
+
+    def get(self, request):
+        q = request.GET.get('q', None)
+
+        plugins = get_plugins()
+        if q:
+            plugins = [v for k, v in plugins.items() if q.casefold() in v['name'].casefold()]
+        else:
+            plugins = [v for k, v in plugins.items()]
+
+        table = CertifiedPluginTable(plugins, user=request.user)
+        table.configure(request)
+
+        # If this is an HTMX request, return only the rendered table HTML
+        if htmx_partial(request):
+            return render(request, 'htmx/table.html', {
+                'table': table,
+            })
+
+        return render(request, 'core/plugin_list.html', {
+            'table': table,
+        })
+
+
+class PluginView(UserPassesTestMixin, View):
+
+    def test_func(self):
+        return self.request.user.is_staff
+
+    def get(self, request, name):
+
+        plugins = get_plugins()
+        plugin = plugins[name]
+
+        table = PluginVersionTable(plugin['versions'], user=request.user)
```

Comment:
The versions currently come out of the API ordered from oldest to newest. Before we display this data, we should either:
- Reverse the array's ordering (lazy way); or
- Sort the array in descending order by `date`